nixos/modules/virtualisation/azure-common.nix

```diff
@@ -13,36 +22,37 @@ in
     };
   };
 
-  imports = [
-    ../profiles/headless.nix
-    ./azure-agent.nix
-  ];
+  imports = [ ./azure-agent.nix ];
 
   config = {
     virtualisation.azure.agent.enable = true;
-
-    boot.kernelParams = [ "console=ttyS0" "earlyprintk=ttyS0" "rootdelay=300" "panic=1" "boot.panic_on_fail" ];
-    boot.initrd.kernelModules = [ "hv_vmbus" "hv_netvsc" "hv_utils" "hv_storvsc" ];
+    services.cloud-init.enable = true;
+    services.cloud-init.network.enable = true;
+    systemd.services.cloud-config.serviceConfig.Restart = "on-failure";
+
+    boot.kernelParams = [
+      "console=ttyS0"
+      "earlyprintk=ttyS0"
+      "rootdelay=300"
+      "panic=1"
+      "boot.panic_on_fail"
+    ];
+    boot.initrd.kernelModules = [
+      "hv_vmbus"
+      "hv_netvsc"
+      "hv_utils"
+      "hv_storvsc"
+    ];
     boot.initrd.availableKernelModules = lib.optionals cfg.acceleratedNetworking mlxDrivers;
 
     # Accelerated networking
     systemd.network.networks."99-azure-unmanaged-devices.network" = lib.mkIf cfg.acceleratedNetworking {
       matchConfig.Driver = mlxDrivers;
       linkConfig.Unmanaged = "yes";
     };
-    networking.networkmanager.unmanaged = lib.mkIf cfg.acceleratedNetworking
-      (builtins.map (drv: "driver:${drv}") mlxDrivers);
-
-    # Generate a GRUB menu.
-    boot.loader.grub.device = "/dev/sda";
-
-    boot.growPartition = true;
-
-    fileSystems."/" = {
-      device = "/dev/disk/by-label/nixos";
-      fsType = "ext4";
-      autoResize = true;
-    };
```

Comment:
I think removing this is a good change for the same reasons you already mentioned:

> * Moved image specific configs (namely file system configs and grub configs) out of `azure-common.nix`, because these settings are only intended for images generated using `azure-image.nix`. This will make `azure-common.nix` more generic so folks generating azure image using other methods (e.g. disko) can leverage it.

I had to override some of these with `mkOverride` in my own code because I used `azure-common.nix`; you change makes it cleaner.

However, this may be a breaking or unexpected change for some users, so you should add a NixOS release notes entry. You can also highlight some of your other improvements from the PR description in there. Probably people will appreciate it!